Sources/Public/BacktraceCrashLoopDetector.swift

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @objc private func loadEvents() {
+        
+        // Cleanup old events - f.e, for multiple usages of detector
+        startupEvents.removeAll()
+        
+        /*
+         - Since detector's DB is relatively small, UserDefaults are a good option here,
+         plus they allow to avoid a headache with reading/writing to/from the custom file.
+
+         - But we should consider shared computers as well - comment from UserDefaults docs:
+            With the exception of managed devices in educational institutions,
+            a user’s defaults are stored locally on a single device,
+            and persisted for backup and restore.
+            To synchronize preferences and other data across a user’s connected devices,
+            use NSUbiquitousKeyValueStore instead.
+         */
+        guard let data = UserDefaults.standard.object(forKey: BacktraceCrashLoopDetector.plistKey) as? Data
+        else { return }
+
+        guard let array = try? PropertyListDecoder().decode([StartUpEvent].self, from: data)
+        else { return }
+                
+        startupEvents.append(contentsOf: array)
+        BacktraceLogger.debug("Events Loaded: \(startupEvents.count)")
+    }
+
+    @objc internal func saveEvents() {
+        let data = try? PropertyListEncoder().encode(startupEvents)
+        UserDefaults.standard.set(data, forKey: BacktraceCrashLoopDetector.plistKey)
+        BacktraceLogger.debug("Events Saved: \(startupEvents.count)")
+    }
+
+    @objc private func reportFilePath() -> String {
+        
+        // Crash Loop Detector considers all other Backtrace modules as potentially dangerous.
+        // Thats why it formats path to PLCrashReporter's report file itself,
+        // for not to use PLCrashReporter's APIs at all
+        
+        let bundleIDBT = Bundle.main.bundleIdentifier ?? ""
+        let appIDPath = bundleIDBT.replacingOccurrences(of: "/", with: "_")
+
+        let paths = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true)
+        let cacheDir = URL(fileURLWithPath: paths.isEmpty ? "" : paths[0])
+
+        let bundleIDPLCR = "com.plausiblelabs.crashreporter.data"
+        let crashReportDir = cacheDir.appendingPathComponent(bundleIDPLCR)
+                                     .appendingPathComponent(appIDPath)
+
+        let reportName = "live_report.plcrash"
+        let reportFullPath = crashReportDir.appendingPathComponent(reportName)
+                                           .absoluteString
+                                           .replacingOccurrences(of: "file://", with: "")
+        BacktraceLogger.debug("reportFullPath: \(reportFullPath)")
+        return reportFullPath
+    }
+
+    @objc private func hasCrashReport() -> Bool {
+        let exists = FileManager.default.fileExists(atPath: reportFilePath())
+        return exists
+    }
+
+    @objc private func addCurrentEvent() {
```

Comment:
what is current and why it is important to point out this is a current event? It is always a current event no matter what happen :P  Maybe renaming it to add or addStartupEvent will make more sense here?

Reply:
hm - basically I agree - name is a bit confusing - polished out